DataStructures/Tasks/DoublyLinkedList.cs

```diff
@@ -0,0 +1,218 @@
+        {
+            if (index < 0 || index >= length)
+            {
+                throw new IndexOutOfRangeException();
+            }
+
+            Node current = head;
+
+            for (int i = 0; i < index; i++)
+            {
+                current = current.Next;
+            }
+
+            return current.Value;
+        }
+
+        public IEnumerator<T> GetEnumerator()
+        {
+            return new DoublyLinkedListEnumerator(head);
+        }
+
+        public void Remove(T item)
+        {
+            Node current = head;
+
+            while (current != null)
+            {
+                if (EqualityComparer<T>.Default.Equals(current.Value, item))
+                {
+                    if (current.Previous != null)
+                    {
+                        current.Previous.Next = current.Next;
+                    }
+                    else
+                    {
+                        head = current.Next;
+                    }
+
+                    if (current.Next != null)
+                    {
+                        current.Next.Previous = current.Previous;
+                    }
+                    else
+                    {
+                        tail = current.Previous;
+                    }
+
+                    length--;
+
+                    break;
+                }
+
+                current = current.Next;
+            }
+        }
+
+        public T RemoveAt(int index)
+        {
+            if (index < 0 || index >= length)
+            {
+                throw new IndexOutOfRangeException();
+            }
+
+            Node current = head;
+
+            for (int i = 0; i < index; i++)
+            {
+                current = current.Next;
+            }
+
+            if (current.Previous != null)
+            {
+                current.Previous.Next = current.Next;
+            }
+            else
+            {
+                head = current.Next;
+            }
+
+            if (current.Next != null)
+            {
+                current.Next.Previous = current.Previous;
+            }
+            else
+            {
+                tail = current.Previous;
+            }
+
+            length--;
+
+            return current.Value;
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        private class Node
+        {
+            public T Value;
+
+            public Node Next;
+
+            public Node Previous;
+        }
+
+        private class DoublyLinkedListEnumerator : IEnumerator<T>
+        {
+            private Node current;
+            private readonly Node head;
+
+            public DoublyLinkedListEnumerator(Node head)
+            {
+                this.head = head;
+                current = null;
+            }
+
+            public T Current => current.Value;
+
+            object IEnumerator.Current => Current;
+
+            public void Dispose() { }
```

Comment:
Should this `Dispose` method be defined? Usually, if you want to use it, you should implement `IDisposable` interface. Let's remove it to due to empty body as well

Reply:
I was trying something else in there, then I gave up so I forgot to remove it.